apps/cnquery/cmd/plugin.go

```diff
@@ -102,6 +102,14 @@ func (c *cnqueryPlugin) RunQuery(conf *run.RunQueryConfig, runtime *providers.Ru
 		}
 	}
 
+	// FIXME: workaround for gcp-snapshot
+	// For a gcp-snapshot asset, we start with a GCP connection.
+	// This get's overriden by a filesystem connection. The fileswystem connection is what we need for the scan
+	// But later, we need the GCP runtime to cleanup the snapshot disk
+	if runtime.Provider.Instance.Name == "gcp" && runtime.Provider.Connection.Name == "filesystem" {
```

Comment:
if we make sure that calling connect for an asset to which we have already connected is a no-op, that issue will be fixed. I feel like this might be needed in other places too